js/ParticleModifiers.js

```diff
@@ -103,3 +107,26 @@ PHOTONS.EvenIntervalIndexModifier.prototype.update = function( particle, target
 
 }
 
+//=======================================
+// LoopingTimeIntervalIndex Modifier
+//=======================================
+
+PHOTONS.LoopingTimeIntervalIndexModifier = function( totalSteps, imagesPerSecond) {
+		PHOTONS.Modifier.call( this );
+		this.totalSteps = Math.floor( totalSteps || 1 );
+		this.timePerImage = 1 / imagesPerSecond;
+	}
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype = Object.create( PHOTONS.Modifier.prototype );
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype.update = function( particle, target ) {
+
+	// keep track of the atlas we started on so we can start randomly and progress normally afterwards
+	if ( particle.atlasStartIndex == undefined ) 
+		particle.atlasStartIndex = particle.atlasIndex.x;	

```

Comment:
Again, I completely agree and hesitantly put that in there after I had a need to maintain which atlas index we started with (in case it was randomized).  What do you recommend?